build-tools/fetch-schemas-plugin.ts

```diff
@@ -0,0 +1,36 @@
+import {remoteSchemas} from "../schemas.config";
+import yaml from 'js-yaml';
+import crypto from 'crypto';
+import fs from "fs";
+
+
+export function vitePluginFetchSchemas() {
+    return {
+        name: 'vite-plugin-fetch-schema',
+        buildStart() {
+            fetchRemoteSchemas();
+        }
+    };
+}
+
+function sha256(data: string): string {
+    return crypto.createHash('sha256').update(data, 'utf8').digest('hex');
+}
+
+export function fetchRemoteSchemas() {
+    remoteSchemas.forEach(async (schema) => {
+        if (!fs.existsSync(`./public/schemas/${schema.name}.json`)) {
+            const fetchedSchemaYAML = await (await (await fetch(schema.url)).blob()).text();
```

Comment:
Is this considered good practice?  Btw, where does the `fetch` come from?

Reply:
@a-w50 you are right, this can be simplified + a .then-chain might be more readable.

Fixed